src/OrchardCore/OrchardCore.ContentManagement/ContentDefinitionManager.cs

```diff
@@ -340,12 +343,26 @@ private ContentDefinitionRecord GetContentDefinitionRecord()
             return record;
         }
 
+        public void Cancel() => _saved?.Clear();
+
         private void UpdateContentDefinitionRecord()
         {
-            var contentDefinitionRecord = LoadContentDefinitionRecord();
+            if (_saved == null)
+            {
+                _saved = new List<bool>() { true };
```

Comment:
Why is it a list?!!

Reply:
Need to be a referenced object (so also a non null one), e.g if you use a bool and change its state after the lambda as been registered, when the lambda will execute it will see the original state.

I could have used a class with the `Save` and `Cancel` properties.

Reply:
Hmm, but because it executes at the end but in the same scope, maybe i will just reference the 
content def manager scoped service itself.

Reply:
Just tried, yes we can pass a simple bool because here no problem that the lambda uses a reference on the service itself. But i need 2 bools, one for registering once, and one for cancelling.

Reply:
Yes, did it to quickly because i was already using instance methods ;)

Reply:
Thanks for the clarification